Mathlib/Algebra/Parity.lean

```diff
@@ -351,6 +352,14 @@ theorem odd_two_mul_add_one (m : α) : Odd (2 * m + 1) :=
   ⟨m, rfl⟩
 #align odd_two_mul_add_one odd_two_mul_add_one
 
+@[simp] lemma odd_add_self_one : Odd (m + m + 1) := ⟨m, by rw [two_mul]⟩
+
+@[simp] lemma odd_add_self_one' : Odd (m + (m + 1)) := by simp [← add_assoc]
+
+@[simp] lemma odd_add_one_self : Odd (m + 1 + m) := by simp [add_comm _ m]
+
+@[simp] lemma odd_add_one_self' : Odd (m + (1 + m)) := by simp [add_comm]
+
```

Comment:
I'm not convinced this is a particularly sensible use of `simp`; clearly we can't scale to `Odd (x + y + 1 + x + y)` etc, and so maybe this is best left to `norm_num`

Reply:
I agree that these lemmas are (at best) on the border of `simp`'s responsibilities.

Personally I am think they're worth keeping but if you're not persuaded, I'll excise them.

Reply:
Is there a reason you didn't include `1 + m + m` and `1 + (m + m)` here?

Reply:
Nope, I should have included them. Now added.